[Incremental Compiler] Add target timeline (#692)
rust-lang/rust-project-goals: src/2026/incremental-system-rethought.md

## Frequently asked questions

## Target timeline

> The duration of the project is 12 months. Starting from the agreed start date ("Month 1"), the target timeline is:

Reviewer: The "over the next 6 months" section above needs to be changed, and this whole timeline should probably be moved there. This goal uses the old format from 2025

*[View changes since the review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/rust-project-goals/692/5c19d5b9a8ccaff909468584fe708acbe08d0aab..d56f521b0a76070c27f8071882d8eb563bc29a8d)*

- Month 1-3:
  - Resolve all concerns, merge RFC. (In parallel with experimentation)
  - Stablish formal communication with the notable teams. (i.e. cargo and compiler)
  - Experiment with `cargo build` building off of `cargo check`.
- Month 3-6:
  - Declare semi-stable fingerprinting format following any known schema.
  - Notion of data dependencies & atomic levels in the compiler.
  - Have some kind of `cargo build` <-> `cargo check` compatibility merged in the compiler.
- Month 6-12:

>   - Implement data dependency difing algorithm in the compilerr for the different stages.

Reviewer: ```suggestion
  - Implement data dependency diffing algorithm in the compiler for the different stages.
```

*[View changes since the review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/rust-project-goals/692/5c19d5b9a8ccaff909468584fe708acbe08d0aab..d56f521b0a76070c27f8071882d8eb563bc29a8d)*

  - Polishing, perf measurements, integration testing...
  - Measure interaction with official rustc-driver projects, such as Clippy or Miri.

>   - (secondary) Measure interaction with unnofficial rustc-driver projects.

Reviewer: ```suggestion
  - (secondary) Measure interaction with unofficial rustc-driver projects.
```

*[View changes since the review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/rust-project-goals/692/5c19d5b9a8ccaff909468584fe708acbe08d0aab..d56f521b0a76070c27f8071882d8eb563bc29a8d)*

> - Month 12-...:

Reviewer: ```suggestion
- Month 12-future:
```

*[View changes since the review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/rust-project-goals/692/5c19d5b9a8ccaff909468584fe708acbe08d0aab..d56f521b0a76070c27f8071882d8eb563bc29a8d)*
